pkg/generators/openapi.go

```diff
@@ -647,14 +647,41 @@ func (g openAPITypeWriter) generateMapProperty(t *types.Type) error {
 	case types.Struct:
 		g.generateReferenceProperty(elemType)
 	case types.Slice, types.Array:
-		g.generateSliceProperty(elemType)
+		if err := g.generateSliceProperty(elemType); err != nil {
+			return err
+		}
+	case types.Map:
+		if err := g.generateNestedMapProperty(t.Name, elemType); err != nil {
```

Comment:
why can't we just call generateMapProperty here?

Reply:
@stts done